chore: rename `node-launchpad` crate to `sn_node_launchpad`

jacderida opened the pull request:
The binary name will be `node-launchpad`, but the crate is named consistently along with the other crates in the workspace (including the use of underscores rather than hyphens).

## Description

reviewpad:summary

grumbach commented on sn_node/src/comm/listener.rs:
```diff
@@ -67,8 +67,6 @@ impl MsgListener {
                     };
 
```
This is now solved!

grumbach commented on sn_node/src/comm/listener.rs:
```diff
@@ -67,8 +67,6 @@ impl MsgListener {
 
                     let src_name = match wire_msg.auth() {
-                        // NB TODO is default acceptable??
-                        AuthKind::SectionShare(_) => Default::default(),
                         AuthKind::Client(auth) => auth.public_key.into(),
                         AuthKind::Node(auth) => {
```
Now I'm happy again 🕺

davidrusu reviewed (approved):
:fire:

davidrusu commented:
bors r+

grumbach commented:
bors r+

jacderida committed:
chore: rename `node-launchpad` crate to `sn_node_launchpad`

The binary name will be `node-launchpad`, but the crate is named consistently along with the other
crates in the workspace (including the use of underscores rather than hyphens).
